### PR TITLE
feat: register-context-provider

### PR DESCRIPTION
**Pull request by Noggling**
## Prerequisites

Please make sure you can check the following boxes:

-   [x] My code follows the code style of this project
-   [x] All new and existing tests passed
-   [ ] Fix any eslint/prettier warnings/errors: npm run lint
-   [ ] Remove unnecessary console.log (use logInfo logWarn if it's needed)
-   [x] Are variable and function names properly describing what it does?

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

-   [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
-   [x] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] I have updated the documentation accordingly
-   [ ] I have added tests to cover my changes

### Description

[Place a meaningful description here]

### Remarks

[Optionally place any follow-up comments, remarks, observations, or notes here for future reference]

### PR Title

Make sure the pull request title follows the conventional commits specification:

-   The format should be \<**type**\>(#UserStoryNumber): \<description\>
-   Use a ! after the type to indicate a breaking change.
-   Example of bad title: "Correct typo"
-   Example of good title: "fix: Correct typo"
-   Some more good titles:
    -   feat: Add support for Node 12.
    -   refactor!: Drop support for Node 6.
    -   feat(ui): Add Button component.
-   Approved types include:
    -   feat: A new feature.
    -   fix: A bug fix.
    -   docs: Documentation only changes.
    -   refactor: A code change that neither fixes a bug nor adds a feature.
    -   perf: A code change that improves performance.
    -   build: Changes that affect the build system or external dependencies.
    -   chore: Other changes that don't modify src or test files.
-   For more information refer to conventionalcommits.org/en/v1.0.0/.

### CODE REVIEW

File: packages/echo-core/README.md [hunk 1]
```diff
@@ -144,3 +193,169 @@ npm unlink
...
 
 _OBS: Errors when switching branches can happen if link is not unlinked properly before linking in the branch switched to. Symlink is unable to find the package after switching branch because the folders don’t exist. When this happens, check out your original branch and start from the beginning with linking the package and the project._
+
+# Echo Modules
+
+## Register application components with Setup
+
+### Register App
+
+Echo is built up around modules, a module can consist of serval applications, It is recommended to keep modules as small as possible preventing long loading times at application/client (Echo) startup. All modules are registered using a setup function exported form the modules entry point. This function is called at Echo startup. The function is called with the EchoModuleApi. The api allows for app registration using `registerApp` as shown below. App registration is used for registration of the main application in a module, based on metadata taken from the modules manifest, the optional parameter can help with configuring the application.
```
Comment (jorgenholme96) [inline packages/echo-core/README.md:203]: serval => several

Comment (jorgenholme96) [reply packages/echo-core/README.md:203]: form => from

File: packages/echo-core/src/types/context.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    return (
+        <Context.Provider>
+            {children}
+        </Context.Provider>
+        );
+};
+ *```
+ **/
+
+export type ContextProvider = React.FunctionComponent<ContextProviderProps>;
```
Comment (Noggling) [inline packages/echo-core/src/types/context.ts:23]: remove this file.

File: packages/echo-core/src/types/index.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 export * from './api';
 export * from './common';
+export * from './context';
```
Comment (Noggling) [inline packages/echo-core/src/types/index.ts:3]: remove

**Review by ingridklepsvik — APPROVED** [review]

**Pull request merged**